testing/databaseTest.py

```diff
@@ -970,6 +976,20 @@ def kinetics_check_reactant_and_product_template(self, family_name):
             if boo:
                 raise ValueError("Error occured in databaseTest. Please check log warnings for all error messages.")
 
+    def kinetics_check_num_reactant_and_product(self, family_name):
+        """
+        This test checks that if the number of reactants and products are specified in the groups.py for
+        rate rules that are generated from the ATG.
+        """
+        family = self.database.kinetics.families[family_name]
+        if family.auto_generated:
```

Comment:
Can you help me understand why we only do this check if the tree was autogenerated? There are many surface reaction families that are missing the `autoGenerated=True` but they still define `reactantNum` and `productNum`. [This](https://github.com/ReactionMechanismGenerator/RMG-database/blob/2118733e3d2b8fcbfc8d23c15f665799b3920e6a/input/kinetics/families/Surface_Dissociation_vdW/groups.py#L21-L24
) is one example. Also halocarbene recombination ([link](https://github.com/ReactionMechanismGenerator/RMG-database/blob/f1a84af193e62347263de66b24802c3075b1b54a/input/kinetics/families/halocarbene_recombination/groups.py#L18-L22
))

I guess once all of our rate rules have been passed through ATG (hopefully in the near future), lines like [this](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/73dcb294fae0201da83407d822dd86fb89fd9411/rmgpy/data/kinetics/family.py#L739) will no longer be necessary since everything will rely on `reactantNum` and `productNum`?